### PR TITLE
Sync chat sample from upstream after `1.4.0` release

### PR DESCRIPTION
**Pull request by prprabhu-ms**
See https://github.com/Azure/communication-ui-library/blob/main/docs/releases/hero-samples.md for details on routine hero sample maintenance on stable version releases of `@azure/communication-react`.

This PR syncs the hero sample from upstream at https://github.com/Azure/communication-ui-library/tree/1.4.0/samples/Chat

### CODE REVIEW

File: Chat/src/app/App.tsx [hunk 1]
```diff
@@ -10,10 +10,17 @@ import { EndScreen } from './EndScreen';
 import { ErrorScreen } from './ErrorScreen';
 import HomeScreen from './HomeScreen';
 import { getExistingThreadIdFromURL } from './utils/getExistingThreadIdFromURL';
+import { getBuildTime, getChatSDKVersion, getCommnicationReactSDKVersion } from './utils/utils';
```
Comment (prprabhu-ms) [inline Chat/src/app/App.tsx]: Remove

File: Chat/src/app/App.tsx [hunk 2]
```diff
@@ -10,10 +10,17 @@ import { EndScreen } from './EndScreen';
...
 import { initializeFileTypeIcons } from '@fluentui/react-file-type-icons';
 
 setLogLevel('warning');
 
+console.info(
```
Comment (prprabhu-ms) [inline Chat/src/app/App.tsx]: Remove

File: Chat/src/app/ChatScreen.tsx [hunk 3]
```diff
@@ -11,7 +11,7 @@ import {
   useAzureCommunicationChatAdapter
 } from '@azure/communication-react';
 import { Stack } from '@fluentui/react';
-import React, { useCallback, useEffect, useMemo } from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
```
Comment (prprabhu-ms) [inline Chat/src/app/ChatScreen.tsx]: Bad conditional compilation?

File: Chat/src/app/PeopleButton.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (prprabhu-ms) [inline Chat/src/app/PeopleButton.tsx]: People button should be beta only, right?

File: Chat/src/app/utils/utils.ts [hunk 5]
```diff
@@ -1,6 +1,13 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+declare let __BUILDTIME__: string; // Injected by webpack
```
Comment (prprabhu-ms) [inline Chat/src/app/utils/utils.ts]: remove

**Review by JamesBurnside — APPROVED** [review]

**Pull request merged**